### PR TITLE
Update Referral Page on Admin

### PR DESCRIPTION
**Pull request by corymcdonald**
## Update Referral Page on Admin

localhost_3000_admin_publishers_e78e8145-9ea3-45b9-abab-3cd5dffded34_referrals

#### Features

- Add last updated field to promo panel
- Add ability to view groupings in admin view

#### How To Test

- Mock data for promo stats by running `rails database_updates:mock_data:populate_promo_stats`
- Login to admin and open up the referrals.
- Ensure data is present and you can change the month

### CODE REVIEW

File: app/controllers/publishers/promo_registrations_controller.rb [hunk 1]
```diff
@@ -1,11 +1,9 @@
...
+  GROUP_START_DATE = Date.new(2019, 10, 1)
+
   def for_referral_code
-    promo_registration =
-      if current_publisher.admin?
-        PromoRegistration.find_by(referral_code: params[:referral_code])
-      else
-        current_publisher.promo_registrations.find_by(referral_code: params[:referral_code])
-      end
+    promo_registration = user.promo_registrations.find_by(referral_code: params[:referral_code])
```
Comment (yachtcaptain23) [inline app/controllers/publishers/promo_registrations_controller.rb]: Use `current_user` which is defined in `ApplicationController`

File: app/helpers/promos_helper.rb [hunk 2]
```diff
@@ -20,8 +20,12 @@ def offline_promo_stats
     { "times" => [Time.now.to_s], "series" => { "name" => "downloads", "values" => [rand(0..1000)] }, "aggregate" => { "downloads" => 200, "finalized" => 30 } }
   end
 
+  def publisher_referrals_last_update(publisher)
+    time_ago_in_words(publisher.promo_registrations.order(updated_at: :desc).first&.updated_at)
+  end
+
   def publisher_referral_totals(publisher)
-    aggregate_stats = PromoRegistration.stats_for_registrations(promo_registrations: publisher.promo_registrations)
+    aggregate_stats ||= PromoRegistration.stats_for_registrations(promo_registrations: publisher.promo_registrations)
```
Comment (yachtcaptain23) [inline app/helpers/promos_helper.rb]: Unnecessary change

File: app/controllers/publishers/promo_registrations_controller.rb (hunk 1 above)
Comment (corymcdonald) [reply app/controllers/publishers/promo_registrations_controller.rb]: This is defined below, it's either the passed in value if the user is an administrator, or the current_user :)

File: app/helpers/promos_helper.rb (hunk 2 above)
Comment (corymcdonald) [reply app/helpers/promos_helper.rb]: https://github.com/brave-intl/publishers/pull/2367/commits/15dc77d7f9445934251851a0c3dbf746c7dab00e
